scalastyle-config.xml

```diff
@@ -0,0 +1,81 @@
+<scalastyle>
+  <name>Scalastyle configuration for Scala.js</name>
+
+  <check level="error" enabled="true" class="org.scalastyle.file.FileTabChecker"/>
+  <check level="error" enabled="true" class="org.scalastyle.file.WhitespaceEndOfLineChecker"/>
+  <check level="error" enabled="true" class="org.scalastyle.file.NewLineAtEofChecker"/>
+
+  <!--Indentation checker, whilst useful gives false positives on the file headers, so disabled for now-->
+  <check level="error" enabled="false" class="org.scalastyle.file.IndentationChecker"/>
+
+  <!--
+  Enforce the hard limit on line length, which is 120 characters.
+  Most of the time the soft limit of 80 characters should be observed, but
+  we can hardly tell an automated style checker about a soft limit.
+  -->
+  <check level="error" enabled="true" class="org.scalastyle.file.FileLineLengthChecker">
+    <parameters>
+      <parameter name="maxLineLength"><![CDATA[120]]></parameter>
+    </parameters>
+  </check>
+
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.RedundantIfChecker"/>
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.SimplifyBooleanExpressionChecker"/>
+
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.UppercaseLChecker"/>
+
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.ProcedureDeclarationChecker"/>
+
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.ClassNamesChecker">
+    <parameters>
+      <parameter name="regex"><![CDATA[[A-Z][A-Za-z]*]]></parameter>
```

Comment:
Can't we have numbers in class names? Like in: `BitSet`, `BitSet1`, `BitSet2`, `BitSetN`